crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/class_info.rs

```diff
@@ -275,6 +279,163 @@ mod get_utility_info_tests {
+        if variant_text.starts_with(format!("{}-[", target).as_str()) {
+            return VariantMatch::Exact;
+        }
+
+        // Check if the variant starts with the (partial) target.
+        if variant_text.starts_with(target) && variant_text != target {
+            return VariantMatch::Partial;
+        }
+        // If all of the above checks fail, there is no match.
+        VariantMatch::None
+    }
+}
+
+#[cfg(test)]
+mod variant_match_tests {
+    use crate::lint::nursery::use_sorted_classes::class_info::VariantMatch;
+
+    #[test]
+    fn test_exact_match() {
+        assert_eq!(VariantMatch::from("hover", "hover"), VariantMatch::Exact);
+        assert_eq!(VariantMatch::from("focus", "focus"), VariantMatch::Exact);
+        assert_eq!(
+            VariantMatch::from("group", "group-[.is-published]"),
+            VariantMatch::Exact
+        );
+        assert_eq!(
+            VariantMatch::from("has", "has-[:checked]"),
+            VariantMatch::Exact
+        );
+        assert_eq!(
+            VariantMatch::from("group-has", "group-has-[.custom-class]"),
+            VariantMatch::Exact
+        );
+        assert_eq!(
+            VariantMatch::from("group-aria-disabled", "group-aria-disabled"),
+            VariantMatch::Exact
+        );
+    }
+
+    #[test]
+    fn test_partial_match() {
+        assert_eq!(
+            VariantMatch::from("group", "group-has-[.custom-class]"),
+            VariantMatch::Partial
+        );
+        assert_eq!(
+            VariantMatch::from("peer", "peer-has-[:checked]"),
+            VariantMatch::Partial
+        );
+    }
+}
+
+fn find_variant_info(config_variants: VariantsConfig, variant_text: &str) -> Option<usize> {
+    let mut variant: &str = "<no match>";
+    let mut match_index: usize = 0;
+    let mut last_size: usize = 0;
+
+    // Iterate over each variant looking for a match.
+    for (index, target) in config_variants.iter().enumerate() {
+        match VariantMatch::from(target, variant_text) {
+            VariantMatch::Exact => {
+                // Exact matches can be returned immediately.
+                return Some(index);
+            }
+            VariantMatch::Partial => {
+                // Multiple partial matches can occur, so we need to keep looking to find
+                // the longest target that matches. For example, if the variant text is
+                // `group-aria-[.custom-class]`, and there are targets like `group` and `group-aria`, we want to
+                // make sure that the `group-aria` target is matched as it is more specific,
+                // so when the target is `group` a Partial match will occur.
+                let target_size = target.chars().count();
+                if target_size > last_size {
+                    variant = target;
+                    match_index = index;
+                    last_size = target_size;
+                }
+            }
+            _ => {}
```

Comment:
Small advice I found useful: don't use `_` inside a `match`, because if the future you add a new variant, you won't get compiler error.